### PR TITLE
Actor Critic Hidden Layers

### PR DESCRIPTION
**Pull request by hades-rp2010**
For #222 
the agent classes include policy_layers and value_layers args, and the corresponding changes in all the algorithms and the tests

### CODE REVIEW

**Comment by hades-rp2010** [comment]
Can u take a look @sampreet-arthi 
Shouldnt be much, just small changes here and there

**Comment by hades-rp2010** [comment]
> Some minor changes here and there. @Sharad24 could you take a look at this too?
> 
> Would be good if you could handle separate issues in different PRs (made this mistake myself a few times). It gets confusing if your PR is a longer one.

Cool, I'll open a new one on Custom_policy docs. Let this one be for the different_layer sizes only

File: genrl/deep/agents/a2c/a2c.py [hunk 1]
```diff
@@ -69,7 +71,8 @@ def __init__(
         gamma: float = 0.99,
         lr_policy: float = 0.01,
         lr_value: float = 0.1,
-        layers: Tuple = (32, 32),
+        policy_layers: Tuple = (32, 32),
+        value_layers: Tuple = (32, 32),
```
[the remark this answers is not in the document]
Comment (hades-rp2010) [reply genrl/deep/agents/a2c/a2c.py:75]: I checked.. they are training

File: docs/source/tutorials/Using Custom Policies.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        self.lstm = nn.LSTM(self.state_dim, layer_size, layers)
+        self.fc = mlp([layer_size] + list(hidden) + [action_dim],
+                      sac=self.sac)  # the mlp layers
+
+    def forward(self, state):
+        state, h = self.lstm(state.unsqueeze(0))
+        state = state.view(-1, self.layer_size)
+        action = self.fc(state)
+        return action
+
```
[the remark this answers is not in the document]
Comment (hades-rp2010) [reply docs/source/tutorials/Using Custom Policies.md:76]: I'll take care of this in the new PR

**Comment by sampreet-arthi** [comment]
If you've resolved any of the issues, feel free to mark them as resolved. If you're still working on it, tag it with a [WIP] or convert it to a draft PR.

**Comment by hades-rp2010** [comment]
I've opened up a new PR for the custom policy docs, so I guess we could merge this?

File: docs/source/tutorials/Using Custom Policies.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+# Define a custom MLP Policy
+class custom_policy(MlpPolicy):
+    def __init__(self, state_dim, action_dim, **kwargs):
+        super(custom_policy, self).__init__(self, state_dim, 
+                                            action_dim, 
+                                            kwargs.get("hidden"))
+        self.state_dim = state_dim
+        self.action_dim = action_dim
```
[the remark this answers is not in the document]
Comment (hades-rp2010) [reply docs/source/tutorials/Using Custom Policies.md]: Took care of this too.. changed it to `hidden = (64, 64)`, and some small changes, it works

**Comment by sampreet-arthi** [comment]
Please check the other PR once. Make the changes here instead.

**Comment by hades-rp2010** [comment]
@sampreet-arthi Could u take a look at this. I made the requested changes, and resolved the conflicts.
Probably merge this if it looks good?

**Comment by hades-rp2010** [comment]
> Looks good. The multi Actor Critic is not necessary here, it's being used in #227

So I should remove the modifications I made to the multiactorcritic agents?

**Comment by sampreet-arthi** [comment]
Yes please

**Comment by sampreet-arthi** [comment]
Ok will merge this if no other changes are needed.

**Pull request merged**